Source/Core/Core/HW/Memmap.cpp

```diff
@@ -394,15 +395,50 @@ void UpdateLogicalMemory(const PowerPC::BatTable& dbat_table)
 
 void DoState(PointerWrap& p)
 {
-  bool wii = SConfig::GetInstance().bWii;
-  p.DoArray(m_pRAM, GetRamSize());
-  p.DoArray(m_pL1Cache, GetL1CacheSize());
+  const u32 current_ram_size = GetRamSize();
+  const u32 current_l1_cache_size = GetL1CacheSize();
+  const bool current_have_fake_vmem = !!m_pFakeVMEM;
+  const u32 current_fake_vmem_size = current_have_fake_vmem ? GetFakeVMemSize() : 0;
+  const bool current_have_exram = !!m_pEXRAM;
+  const u32 current_exram_size = current_have_exram ? GetExRamSize() : 0;
+
+  u32 state_ram_size = current_ram_size;
+  u32 state_l1_cache_size = current_l1_cache_size;
+  bool state_have_fake_vmem = current_have_fake_vmem;
+  u32 state_fake_vmem_size = current_fake_vmem_size;
+  bool state_have_exram = current_have_exram;
+  u32 state_exram_size = current_exram_size;
+
+  p.Do(state_ram_size);
+  p.Do(state_l1_cache_size);
+  p.Do(state_have_fake_vmem);
+  p.Do(state_fake_vmem_size);
+  p.Do(state_have_exram);
+  p.Do(state_exram_size);
+
+  // If we're loading a savestate and any of the above differs between the savestate and the current
+  // state, cancel the load. This is technically possible to support but would require a bunch of
+  // reinitialization of things that depend on these.
+  if (current_ram_size != state_ram_size || current_l1_cache_size != state_l1_cache_size ||
```

Comment:
Wouldn't be easier to read this way:
```c++
// I created intermediate variables for clarity
const auto current_settings = std::tie(current_ram_size, current_l1_cache_size, current_have_fake_vmem, current_fake_vmem_size, current_have_exram, current_exram_size);
const auto state_settings = std::tie(state_ram_size, state_l1_cache_size, state_have_fake_vmem, state_fake_vmem_size, state_have_exram, state_exram_size);
if (!(current_settings == state_settings))  // != for tuple is removed? in C++20
```

Reply:
operator!= still exists in C++20, it's generated from the use of the `<=>` comparison operator in tuple's interface (which makes all of the manual comparison implementations obsolete. If they actually removed `operator!=`, that would break a ton of code

Reply:
Oh, I see. That was the part I didn't understand correctly, thanks.

Reply:
I guess that's slightly more readable? Don't really mind either way.